internal/provider/organization_resource.go

```diff
@@ -260,3 +473,78 @@ func (r *OrganizationResource) ImportState(ctx context.Context, req resource.Imp
 	// set the `name` attribute.
 	resource.ImportStatePassthroughID(ctx, path.Root("name"), req, resp)
 }
+
+func (r *OrganizationResource) patchGroupSync(
+	ctx context.Context,
+	orgID uuid.UUID,
+	groupSyncObject types.Object,
+) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	// Read values from Terraform
+	var groupSyncData GroupSyncModel
+	diags.Append(groupSyncObject.As(ctx, &groupSyncData, basetypes.ObjectAsOptions{})...)
+	if diags.HasError() {
+		return diags
+	}
+
+	// Convert that into the type used to send the PATCH to the backend
+	var groupSync codersdk.GroupSyncSettings
+	groupSync.Field = groupSyncData.Field.ValueString()
+	groupSync.RegexFilter = regexp.MustCompile(groupSyncData.RegexFilter.ValueString())
```

Comment:
Keep this using `MustCompile`, but add a custom regex validator on each `regex_filter` attribute so it can be caught by `terraform validate`.